frame/composable-traits/src/staking/rewards.rs

```diff
@@ -0,0 +1,26 @@
+use crate::time::{DurationSeconds, Timestamp};
+use codec::{Decode, Encode};
+use composable_support::collections::vec::bounded::BiBoundedVec;
+
+use core::fmt::Debug;
+use frame_support::{storage::bounded_btree_map::BoundedBTreeMap, RuntimeDebug};
+use scale_info::TypeInfo;
+use sp_runtime::Perbill;
+
+pub type DurationMultiplierRewardsConfig<Limit> = BoundedBTreeMap<DurationSeconds, Perbill, Limit>;
+
+#[derive(RuntimeDebug, PartialEq, Eq, Clone, Encode, Decode, TypeInfo)]
+pub struct RewardConfig<RewardsRate> {
+	/// if not update, stops inflating rewards
+	pub end_block: Timestamp,
+	/// example, asset id, amount, and release frequency   
+	pub reward_rates: BiBoundedVec<RewardsRate, 1, 16>,
+}
+
+#[derive(Debug, PartialEq, Eq, Copy, Clone, Encode, Decode, TypeInfo)]
+pub struct Rewards<RewardsUpdates> {
+	/// List of reward asset/pending rewards.
+	pub rewards: RewardsUpdates,
+	/// The reward multiplier. Captured from configuration  on creation.
+	pub reward_multiplier: Perbill,
+}
```

Comment:
I wonder what causes deviation from the data structures mentioned in the design https://github.com/ComposableFi/composable/blob/b0d739202183fabf1c653d100da7f215d6dfdfed/rfcs/0004-pablo-distribution.md?plain=1#L518 ? It would be helpful to know why those would not work.

Reply:
I cannot put proposed rust design onto staking. We can have quick check in the morning.

Reply:
But I would suggest split document into 2, mechanic/protocol design and implementation.